tools/universe/package_publisher.py

```diff
@@ -11,27 +11,76 @@
 log = logging.getLogger(__name__)
 logging.basicConfig(level=logging.DEBUG, format="%(message)s")
 
+# By default, publish GA releases at 0, 100, 200, 300, ... to allow hotfixes within releases.
+ga_index_multiplier = 100
+# Don't bother with skipping indexes for beta releases, just go sequentially.
+beta_index_multiplier = 1
+
 class UniversePackagePublisher(object):
     '''Creates a PR for a release against the Universe repository at http://github.com/mesosphere/universe.
     '''
 
-    def __init__(self, package_name, package_version, commit_desc, dry_run=False):
+    def __init__(self, package_name, package_version, commit_desc, beta_release, dry_run=False):
         self._pkg_name = package_name
         self._pkg_version = package_version
         self._pr_title = 'Release {} {} (automated commit)\n\n'.format(self._pkg_name, self._pkg_version)
         self._commit_desc = commit_desc
+        self._beta_release = beta_release
         self._dry_run = dry_run
 
         # Optional configuration via envvars:
         self._release_branch = os.environ.get('RELEASE_BRANCH', 'version-3.x')
+        self._release_index = int(os.environ.get('RELEASE_INDEX', -1))
         self._github_user = os.environ.get('GITHUB_USER', 'mesosphere-ci')
         self._github_token = os.environ.get('GITHUB_TOKEN', None)
         if self._github_token is None:
-            raise Exception('GITHUB_TOKEN is required: Credential to create a PR against Universe')
+            if self._dry_run:
+                self._github_token = 'DRY_RUN'
+            else:
+                raise Exception('GITHUB_TOKEN is required: Credential to create a PR against Universe')
         self._enc_github_token = base64.encodestring(self._github_token.encode('utf-8')).decode('utf-8').rstrip('\n')
         self._release_universe_repo = os.environ.get('RELEASE_UNIVERSE_REPO', 'mesosphere/universe')
 
 
+    def _find_release_index(self, repo_pkg_base):
+        '''Returns the correct number/id for this release in the universe tree, and the prior release to diff against.
+
+        Returns a tuple containing two ints: [prior_index (or -1 if N/A), this_index]'''
+
+        # Find the index to be used for this release:
+        last_index = -1
+        if self._release_index >= 0:
+            if os.path.exists(os.path.join(repo_pkg_base, str(self._release_index))):
```

Comment:
If we allow for indices to be specified explicitly, then even the automatic case could result in an overlap with an existing release. Should this check not be in the general path?

Reply:
The automatic case is checking what's currently present in the repo to figure out the next number to use as 'latest'. Shouldn't have a conflict assuming things are added sequentially, i.e. unless someone has two parallel PRs out at the same time.